pyproject.toml

```
@@ -1,87 +1,56 @@
 [build-system]
-requires = ["hatchling>=1.4.0", "jupyterlab==3.*", "hatch-nodejs-version"]
+requires = ["hatchling>=1.5.0", "jupyterlab>=4.0.0,<5", "hatch-nodejs-version>=0.3.2"]
 build-backend = "hatchling.build"
 
 [project]
 name = "ipylgbst"
 readme = "README.md"
-license = { file = "LICENSE.txt" }
-requires-python = ">=3.7"
+license = { file = "LICENSE" }
+requires-python = ">=3.8"
 classifiers = [
     "Framework :: Jupyter",
-    "Intended Audience :: Developers",
-    "Intended Audience :: Science/Research",
+    "Framework :: Jupyter :: JupyterLab",
+    "Framework :: Jupyter :: JupyterLab :: 4",
+    "Framework :: Jupyter :: JupyterLab :: Extensions",
+    "Framework :: Jupyter :: JupyterLab :: Extensions :: Prebuilt",
     "License :: OSI Approved :: BSD License",
     "Programming Language :: Python",
     "Programming Language :: Python :: 3",
-    "Programming Language :: Python :: 3.7",
     "Programming Language :: Python :: 3.8",
     "Programming Language :: Python :: 3.9",
     "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
 ]
 dependencies = [
-    "ipywidgets>=7.0.0",
 ]
 dynamic = ["version", "description", "authors", "urls", "keywords"]
 
```

The build is failing because the optional dependencies for `docs` and `test` were removed, but they are still needed. With this command from the build script you are installing them ` python -m pip install --upgrade -v -e ".[test, examples, docs]"`.

```suggestion
[project.optional-dependencies]
docs = [
    "jupyter_sphinx",
    "nbsphinx",
    "nbsphinx-link",
    "pypandoc",
    "pytest_check_links",
    "recommonmark",
    "sphinx>=1.5",
    "sphinx_rtd_theme",
]
examples = []
test = [
    "nbval",
    "pytest-cov",
    "pytest>=6.0",
]
```

---

The suggestion was meant to be adding to `pyproject.toml`, not removing anything, sorry.

---

Thanks Denisa. This makes sense, I did not properly checked that the script to update to jupyter lab 4 was removing these dependencies in the `pyproject.toml`.